crates/utils/src/errors.rs

```diff
@@ -0,0 +1,54 @@
+use std::convert::TryFrom;
+use std::fmt;
+use std::fmt::Formatter;
+use wasm_bindgen::{JsCast, JsValue};
+
+/// Wrapper type around [`js_sys::Error`]
+///
+/// [`Display`][fmt::Display] impl returns the result `error.toString()` from JavaScript
+pub struct JsError {
+    /// `name` from [`js_sys::Error`]
+    pub name: String,
+    /// `message` from [`js_sys::Error`]
+    pub message: String,
+    js_to_string: String,
+}
+
+impl From<js_sys::Error> for JsError {
+    fn from(error: js_sys::Error) -> Self {
+        JsError {
+            name: String::from(error.name()),
+            message: String::from(error.message()),
+            js_to_string: String::from(error.to_string()),
+        }
+    }
+}
+
+/// The [`JsValue`] is not a JsvaScript's `Error`.
+pub struct NotJsError;
+
+impl TryFrom<JsValue> for JsError {
+    type Error = NotJsError;
+
+    fn try_from(value: JsValue) -> Result<Self, Self::Error> {
+        match value.dyn_into::<js_sys::Error>() {
+            Ok(error) => Ok(JsError::from(error)),
+            Err(_) => Err(NotJsError),
+        }
+    }
+}
```

Comment:
```suggestion
impl TryFrom<JsValue> for JsError {
    type Error = JsValue;

    fn try_from(value: JsValue) -> Result<Self, Self::Error> {
        value.dyn_into::<js_sys::Error>().map(JsError::from)
    }
}
```
The `NotJsError` doesn't really give you much and you lose the `JsValue` when calling `JsError::try_from`, chances are most of the time this won't matter but I think this change gives more flexibility?

Reply:
How about keeping `NotJsValue` but giving it access to the `JsValue` and implement std `Error` for it? It can also be used with anyhow that way

Reply:
Sounds good to me :)